File: branches/beta/perl/handler.pl
#!/usr/bin/perl

use CGI;
use CGI::Carp qw(warningsToBrowser fatalsToBrowser); 
use Config::IniFiles;
use File::Spec;
use GrabzIt::GrabzItClient;

#This Perl file handles the GrabzIt callback

print "HTTP/1.0 200 OK";
print "Content-type: text/html\r\n\r\n";

$cgi = new CGI;

my $ConfigFile = 'config.ini';
tie my %ini, 'Config::IniFiles', (-file => $ConfigFile);
my %Config = %{$ini{"GrabzIt"}};

$message = $cgi->param("message");
$customId = $cgi->param("customid");
$id = $cgi->param("id");
$filename = $cgi->param("filename");

#Custom id can be used to store user ids or whatever is needed for the later processing of the
#resulting screenshot

$grabzIt = new GrabzItClient($Config{applicationKey}, $Config{applicationSecret});
$result = $grabzIt->GetPicture($id);

if ($result)
{
	#Ensure that the application has the correct rights for this directory.
	open FILE, ">".File::Spec->catfile("results",$filename) or die $!; 
	binmode FILE;
	print FILE $result; 
	close FILE;
}
1;

File: handler.php
<?php
include("lib/GrabzItClient.class.php");
include("config.php");

//This PHP file handles the GrabzIt callback

$message = $_GET["message"];
$customId = $_GET["customid"];
$id = $_GET["id"];
$filename = $_GET["filename"];
$format = $_GET["format"];

//Custom id can be used to store user ids or whatever is needed for the later processing of the
//resulting screenshot

$grabzIt = new GrabzItClient($grabzItApplicationKey, $grabzItApplicationSecret);
$result = $grabzIt->GetResult($id);

if (!$result)
{
   return;
}

//Ensure that the application has the correct rights for this directory.
file_put_contents("results" . DIRECTORY_SEPARATOR . $filename, $result);
?>